src/types.ts

```diff
@@ -289,6 +313,13 @@ export interface MetaTransactionV1QuoteParams extends SwapQuoteParamsBase {
     takerAddress: string;
 }
 
+/**
+ * Parameters for the V2 Meta Transaction Service price and quote functions.
+ */
+export interface MetaTransactionV2QuoteParams extends MetaTransactionV1QuoteParams {
```

Comment:
why is there a MetaTransactionV2QuoteParams and MetaTransactionsV2QuoteRequestParams? Seems like you should only need one or the other.

Reply:
`MetaTransactionsV2QuoteRequestParams` is the type after parsing the incoming request to `/quote` (same for `MetaTransactionsV1QuoteRequestParams`). 
`MetaTransactionV2QuoteParams` is the type being passed to  `getMetaTransactionV2QuoteAsync` of `MetaTransactionService` (same for `MetaTransactionV1QuoteParams`).

The intention here is to keep it consistent with V1 types.

I think the intention on why V1 is structured this way is to have a type that purely represents the incoming request (`QuoteRequestParams`). 
Once we have the parsed request, we can do some processing and fields to it and thus another type (`QuoteParams`). The new type represents a process / modified object from the incoming request which is not changed, thus the distinction of the 2 types.